### ISSUE TITLE
There is no mechanism to register custom loops anymore since the 2021 Q4 release

### ISSUE THREAD

**Issue by KlemenDEV** [google/blockly#5819]
**Describe the bug**

There is no mechanism to register custom loops anymore since the 2021 Q4 release.

**To Reproduce**

Before, one could do this:

`Blockly.Constants.Loops.CONTROL_FLOW_IN_LOOP_CHECK_MIXIN.LOOP_TYPES.push('my_loop');`

Now after #5730, there is no mechanism to register loops. In blockly_compressed.js, there seems to be no reference of this PR and also doing this:

`Blockly.loopMixin.CONTROL_FLOW_IN_LOOP_CHECK_MIXIN.LOOP_TYPES.push('my_loop');`

throws

`TypeError: undefined is not an object (evaluating 'Blockly.loopMixin.CONTROL_FLOW_IN_LOOP_CHECK_MIXIN')`

**Comment by gonfunko** [comment]
I believe the intended mechanism now is to update Blockly.blocks.loops.loopTypes per https://github.com/google/blockly/blob/master/blocks/loops.js#L284-L305. Can you try that and verify if it works for you?

**Comment by KlemenDEV** [comment]
In this case, the error is:

`TypeError: undefined is not an object (evaluating 'Blockly.blocks.loops')`

**Comment by Taikono-Himazin** [comment]
I'm in a similar situation.

Probably due to this change.
https://github.com/google/blockly/pull/5730

I read the compiled file (2021Q4 blockly blocks python each compressed) and searched for Blockly.blocks.loops.loopTypes
However, Blockly.blocks was undefined, and although Blockly.Blocks existed, loops and loopTypes were not found.

I want to know alternatives.

**Comment by KlemenDEV** [comment]
Indeed it looks like the loops are not registered at all in blockly_compressed and blocks_compressed

**Comment by alschmiedt** [comment]
Hello and thanks for the issue!

#5730 was reverted in #5740. It looks like the way we recommend in the [comment of this function](https://github.com/google/blockly/blob/master/blocks/loops.js#L284-L305) is not working quite like we intended. 

We are currently looking into it and will get back to you.

**Comment by rachel-fenichel** [comment]
And a more general note: As we work to modernize and modularize the core Blockly library, we will probably run into more things like this. If something you're using disappears and we don't have a suggested workaround, please continue to let us know so we can fix it.

**Comment by KlemenDEV** [comment]
This is understandable as with all major refactors. Will let you know if we run into something similar :)

**Comment by KlemenDEV** [comment]
Any updates on this, or maybe anyone found any potential workarounds?

**Comment by alschmiedt** [comment]
I don't have an update for you yet. The main person working on this has been out sick for the last couple of days and quite a few people are out for the holidays. I will let you know if we make any progress on this, but just a heads up it is very likely this won't happen until the new year. 

Sorry for the delay and thanks for your patience : )

**Comment by cpcallen** [comment]
Hi @KlemenDEV!  My apologies for not getting back to you sooner—and thanks for your patience.

I have made a proposal in #5875 to do several things, including re-exporting the exports of `Blockly.blocks.loops` (and in particular the `loopTypes` export) from `Blockly.blocks.all`; this will make `loopTypes` available as a property on the object obtained via `import loops from `blockly/blocks`.

At the moment `loopTypes` is an array of block type name strings, but given how it is used I am minded to change the code to make it a Set instead.  This would allow new loop types to be registered as follows:

```JS
import loops from 'blockly/blocks';

loops.loopTypes.add('my_loop');
```

I will prepare a PR.  Comments here about the proposed interface—or on the PR about the actual implementation—are most welcome.

**Comment by KlemenDEV** [comment]
The API seems clean and good in my book :)

**Comment by alschmiedt** [comment]
@KlemenDEV @Taikono-Himazin 

This has been released in the most [recent patch](https://github.com/google/blockly/releases/tag/7.20211209.3) and is currently available on npm.

**Comment by Taikono-Himazin** [comment]
This bug was fixed in 7.20211209.3, but the change was undone in 7.20211 209.4. I expected 7.20211 209.5 to be released soon, but it hasn't come out yet. Will 7.20211209.5 be released? Or is it better to wait for the next major release?

**Comment by Taikono-Himazin** [comment]
Here, it was written that it works with `Blockly.blocks.all.loops.loopTypes.add ('my_loop') ;`, but changed from `Blockly.blocks.all` to `Blockly.blocks.libraryBlocks` in the release notes. It was stated that it was done.

I confirmed that it was resolved in [8.0.0](https://github.com/google/blockly/releases/tag/8.0.0).
Here is the code that works as a memorandum.
`Blockly.libraryBlocks.loops.loopTypes.add('custom_loop')`
